tests/green_mode/filename_operationsTest.py

```diff
@@ -0,0 +1,111 @@
+        with file_path.open() as stream:
+            contents = yaml.load(stream)
+        ret_val_contents = check_filename_prefix_postfix(deepcopy(contents),
+                                                         3, 3)
+        test_contents = deepcopy(append_to_contents(deepcopy(contents),
+                                                    'filename_prefix',
+                                                    ['example_file_']))
+        test_contents = append_to_contents(deepcopy(test_contents),
+                                           'filename_suffix',
+                                           [''])
+        self.assertEqual(test_contents, ret_val_contents)
+
+        # Works well when min files and min length of prefix are sufficiently
+        # low.
+        contents = {'dir_structure': ['py_some_name.xyz', 'py_some_other.c',
+                                      'py_yet_another_file.yaml']}
+        ret_val_contents = check_filename_prefix_postfix(deepcopy(contents),
+                                                         2, 2)
+        test_contents = deepcopy(append_to_contents(deepcopy(contents),
+                                                    'filename_prefix',
+                                                    ['py_']))
+        test_contents = append_to_contents(deepcopy(test_contents),
+                                           'filename_suffix',
+                                           [''])
+        self.assertEqual(test_contents, ret_val_contents)
+
+        # Works when min length of prefix and min files is exactly equal
+        # to the paramters passed.
+        ret_val_contents = check_filename_prefix_postfix(deepcopy(contents),
+                                                         3, 3)
+        test_contents = deepcopy(append_to_contents(deepcopy(contents),
+                                                    'filename_prefix',
+                                                    ['py_']))
+        test_contents = append_to_contents(deepcopy(test_contents),
+                                           'filename_suffix',
+                                           [''])
+        self.assertEqual(test_contents, ret_val_contents)
+
+        # Doesn't work when min length of prefix exceeds the prefix value.
+        ret_val_contents = check_filename_prefix_postfix(deepcopy(contents),
+                                                         4, 3)
+        test_contents = deepcopy(append_to_contents(deepcopy(contents),
+                                                    'filename_prefix',
+                                                    ['']))
+        test_contents = append_to_contents(deepcopy(test_contents),
+                                           'filename_suffix',
+                                           [''])
+        self.assertEqual(test_contents, ret_val_contents)
+
+        # Doesn't work when min files exceed the files for a given prefix.
```

Comment:
This means it only works when both the values are equal, right?

Reply:
when the length of prefix is greater than or equal to minimum length of prefix and when the number of files matched against that prefix is greater than or equal to the `min_files_for_prefix`, it works 😄

Reply:
both conditions must satisfy and the values should be greater than the allowed values in both cases for it to consider a detected prefix for testing it as a value against the bear